api/user.go

```diff
@@ -443,6 +443,37 @@ func LoginByEmail(c *Context, w http.ResponseWriter, r *http.Request, email, nam
 	return nil
 }
 
+func LoginByUsername(c *Context, w http.ResponseWriter, r *http.Request, username, name, password, deviceId string) *model.User {
+	var team *model.Team
+
+	if result := <-Srv.Store.Team().GetByName(name); result.Err != nil {
+		c.Err = result.Err
+		return nil
+	} else {
+		team = result.Data.(*model.Team)
+	}
+
+	if result := <-Srv.Store.User().GetByUsername(team.Id, username); result.Err != nil {
+		c.Err = result.Err
+		c.Err.StatusCode = http.StatusForbidden
+		return nil
+	} else {
+		user := result.Data.(*model.User)
+
+		if len(user.AuthData) != 0 {
+			c.Err = model.NewAppError("LoginByUsername", "Please sign in using "+user.AuthService, "")
```

Comment:
All errors should now be using `NewLocAppError` and all logging should use `utils.T()`